rdmo/projects/filters.py

```diff
@@ -13,6 +18,88 @@ class Meta:
         fields = ('title', 'catalog')
 
 
+class ProjectSearchFilterBackend(SearchFilter):
+
+    def filter_queryset(self, request, queryset, view):
+        if view.detail:
+            return queryset
+
+        search_terms = self.get_search_terms(request)
+        if search_terms:
+            for search_term in search_terms:
+                queryset = queryset.filter(
+                    Q(title__icontains=search_term) | (
+                        Q(memberships__role='owner') & (
+                            Q(memberships__user__username__icontains=search_term) |
+                            Q(memberships__user__first_name__icontains=search_term) |
+                            Q(memberships__user__last_name__icontains=search_term) |
+                            Q(memberships__user__email__icontains=search_term)
+                        )
+                    )
+                )
+
+        return queryset
+
+
+class ProjectDateFilterBackend(BaseFilterBackend):
```

Comment:
This allows for http://localhost:8000/api/v1/projects/projects/?after=2017-01-30&before=2020-01-30T0:00:00, with `before` and `after` or one of those.